components/brave_wallet/browser/buy_and_sell_response_parser.cc

```diff
@@ -0,0 +1,520 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/buy_and_sell_response_parser.h"
+
+#include <string>
+#include <utility>
+#include <vector>
+#include "brave/components/brave_wallet/common/brave_wallet.mojom-forward.h"
+
+namespace {
+
+bool ParseMeldLogos(const base::Value::Dict* logos,
+                    brave_wallet::mojom::LogoImages* logo_images) {
+  if (!logos) {
+    return false;
+  }
+  if (const auto* dark_logo = logos->FindString("dark")) {
+    logo_images->dark_url = *dark_logo;
+  }
+  if (const auto* dark_short_logo = logos->FindString("darkShort")) {
+    logo_images->dark_short_url = *dark_short_logo;
+  }
+  if (const auto* light_logo = logos->FindString("light")) {
+    logo_images->light_url = *light_logo;
+  }
+  if (const auto* light_short_logo = logos->FindString("lightShort")) {
+    logo_images->light_short_url = *light_short_logo;
+  }
+
+  return true;
+}
+
+}  //  namespace
+
+namespace brave_wallet {
+bool ParseServiceProviders(
+    const base::Value& json_value,
+    std::vector<mojom::ServiceProviderPtr>* service_providers) {
+  // Parses results like this:
+  // {
+  //    "categories": [ "CRYPTO_ONRAMP" ],
+  //    "categoryStatuses": {
+  //       "CRYPTO_ONRAMP": "LIVE"
+  //    },
+  //    "logos": {
+  //       "dark": "https://images-serviceprovider.meld.io/BANXA/logo_dark.png",
+  //       "darkShort":
+  //       "https://images-serviceprovider.meld.io/BANXA/short_logo_dark.png",
+  //       "light":
+  //       "https://images-serviceprovider.meld.io/BANXA/logo_light.png",
+  //       "lightShort":
+  //       "https://images-serviceprovider.meld.io/BANXA/short_logo_light.png"
+  //    },
+  //    "name": "Banxa",
+  //    "serviceProvider": "BANXA",
+  //    "status": "LIVE",
+  //    "websiteUrl": "http://www.banxa.com"
+  // }
+
+  DCHECK(service_providers);
+
+  if (!json_value.is_list()) {
+    LOG(ERROR) << "Invalid response, could not parse JSON, JSON is not a list";
+    return false;
+  }
+
+  for (const auto& sp_item : json_value.GetList()) {
+    if (!sp_item.is_dict()) {
+      LOG(ERROR)
+          << "Invalid response, could not parse JSON, JSON is not a dict";
+      return false;
+    }
+
+    auto sp = mojom::ServiceProvider::New();
+    const std::string* sp_name = sp_item.GetDict().FindString("name");
+    if (!sp_name) {
+      return false;
+    }
+    sp->name = *sp_name;
```

Comment:
Let's use idl-generated parsers like here https://github.com/brave/brave-core/blob/master/components/brave_wallet/browser/swap_response_parser.cc#L109
for all parsings in this new file